p4/include/tps_headers.p4

```diff
@@ -134,14 +140,15 @@ struct metadata {
 }
 
 struct headers {
-    ethernet_t     ethernet;
-    ipv4_t         ipv4;
-    ipv6_t         ipv6;
-    int_ip_shim_t  int_shim;
-    int_header_t   int_header;
+    ethernet_t        ethernet;
+    ipv4_t            ipv4;
+    ipv6_t            ipv6;
+    udp_t             udp_int;
+    int_udp_shim_t    int_shim;
+    int_header_t      int_header;
+    int_metadata_t    int_meta_3;
+    int_metadata_t    int_meta_2;
     source_metadata_t int_meta;
-    int_metadata_t int_meta_2;
-    int_metadata_t int_meta_3;
-    udp_t          udp;
-    tcp_t          tcp;
+    udp_t             udp;
+    tcp_t             tcp;
```

Comment:
Why parse the tcp header?

We should just need to know the IP protocol for all non UDP packets.

Reply:
for gateway